src/reactor/kqueue.rs

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::os::kqueue::Signal;
+
+use polling::os::kqueue::{PollerKqueueExt, Process, ProcessOps, Signal as PollSignal};
+use polling::{Event, PollMode, Poller};
+
+use std::fmt;
+use std::io::Result;
+use std::os::unix::io::RawFd;
+use std::process::Child;
+
+/// The raw registration into the reactor.
+///
+/// This needs to be public, since it is technically exposed through the `QueueableSealed` trait.
+#[doc(hidden)]
+pub enum Registration {
+    /// Raw file descriptor for readability/writability.
+    Fd(RawFd),
+
+    /// Raw signal number for signal delivery.
+    Signal(Signal),
+
+    /// Process for process termination.
+    Process(Child),
+}
+
+impl fmt::Debug for Registration {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Fd(raw) => fmt::Debug::fmt(raw, f),
+            Self::Signal(signal) => fmt::Debug::fmt(signal, f),
+            Self::Process(process) => fmt::Debug::fmt(process, f),
+        }
+    }
+}
+
+impl From<RawFd> for Registration {
+    fn from(raw: RawFd) -> Self {
+        Self::Fd(raw)
+    }
+}
+
+impl From<Signal> for Registration {
+    fn from(signal: Signal) -> Self {
+        Self::Signal(signal)
+    }
+}
+
+impl From<Child> for Registration {
+    fn from(process: Child) -> Self {
+        Self::Process(process)
+    }
+}
```

Comment:
is that really necessary?

Reply:
The first one is necessary, since `RawFd` needs to implement `Into<Registration>`. The others, not so much. Fixed!